### PR TITLE
Handle Vue slot shorthand syntax

### LINKED ISSUE #289: Bug: Can not handle Vue slot shortcut syntax
### Plugin Version

master

### Prettier Version

^2.3

### Which frameworks are affected?

- [ ] none
- [X] vue
- [ ] angular
- [ ] svelte

### Node Version

14

### Which operating systems have you used?

- [ ] Linux
- [ ] macOS
- [X] Windows

### Prettier config

```json
{
  printWidth: 120,
  singleQuote: true,
  trailingComma: 'all',
  pugAttributeSeparator: 'none',
  pugFramework: 'vue',
  pugSingleFileComponentIndentation: true,
  pugSingleQuote: false,
}
```

### Input

```pug
my-component(foo="bar", #default="{ baz }")
```

### Output or Error

```bash
my-component(foo="bar", #default="{ baz }")
```

### Expected Output

```pug
my-component(foo="bar" #default="{ baz }")
```

### Additional Context

it just does not format the content at all (via VS code). removing the destructuring will make it work

### PR DESCRIPTION
**Pull request by Shinigami92**
close #289

### CODE REVIEW

File: tests/issues/issue-289/unformatted.pug [hunk 1]
```diff
@@ -0,0 +1 @@
+my-component( foo="bar",  #default="{  baz }" )
```
Comment (Shinigami92) [inline tests/issues/issue-289/unformatted.pug:1]: @backbone87 Please provide an example that fails the test 🤔

Comment (Shinigami92) [reply tests/issues/issue-289/unformatted.pug:1]: If you're using Vue 3, could you try to remove/disable `pugSingleFileComponentIndentation`? Just guessing that maybe there is a problem with that?! :thinking: But not sure.

Comment (backbone87) [reply tests/issues/issue-289/unformatted.pug:1]: failing:

unformatted: `my-component(  #default="{ foo, bar: baz }")`
formatted: `my-component(#default="{ foo, bar: baz }")`

```
SyntaxError: Unexpected token ','

      301 |                             }
      302 |                     } catch (error: any) {
    > 303 |                             throw new Error(error);
          |                                   ^
      304 |                     }
      305 |                     token = this.getNextToken();
      306 |             }
```

Comment (backbone87) [reply tests/issues/issue-289/unformatted.pug:1]: the test passes with the following change:
```ts
export function isVueExpression(name: string): boolean {
	return /^((v-(bind|slot))?:|v-(model|slot|if|for|else-if|text|html)|#).*/.test(name);
}
```
(added `|#`)

**Pull request merged**